Scripts/Editor/Debug/ScriptDefinesToggle.cs

```diff
@@ -0,0 +1,216 @@
+using Anvil.CSharp.Logging;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using UnityEditor;
+using UnityEditor.Build;
+using UnityEditor.Callbacks;
+
+namespace Anvil.Unity.Editor.Debug
+{
+    /// <summary>
+    /// Editor helper class to easily toggle on and off Script Defines and their dependencies.
+    /// </summary>
+    [InitializeOnLoad]
+    public static class ScriptDefinesToggle
+    {
+        /// <summary>
+        /// The base location for where items should appear in the menu.
+        /// </summary>
+        public const string MENU_PATH_BASE = "Anvil/Script Defines";
+
+        private static readonly NamedBuildTarget NAMED_BUILD_TARGET;
+        private static readonly Dictionary<string, ScriptDefineDefinition> SCRIPT_DEFINE_DEFINITIONS_LOOKUP;
+        private static readonly Logger LOGGER;
+
+        static ScriptDefinesToggle()
+        {
+            LOGGER = Log.GetStaticLogger(typeof(ScriptDefinesToggle));
+            SCRIPT_DEFINE_DEFINITIONS_LOOKUP = new Dictionary<string, ScriptDefineDefinition>();
+
+            BuildTarget buildTarget = EditorUserBuildSettings.activeBuildTarget;
+            BuildTargetGroup buildTargetGroup = BuildPipeline.GetBuildTargetGroup(buildTarget);
+            NAMED_BUILD_TARGET = NamedBuildTarget.FromBuildTargetGroup(buildTargetGroup);
+        }
+
+        [DidReloadScripts]
+        private static void ValidateSafety()
+        {
+            PlayerSettings.GetScriptingDefineSymbols(NAMED_BUILD_TARGET, out string[] defines);
+            HashSet<string> definesHashSet = defines.ToHashSet();
+
+            foreach (ScriptDefineDefinition scriptDefineDefinition in SCRIPT_DEFINE_DEFINITIONS_LOOKUP.Values)
+            {
+                ValidateForErrors(scriptDefineDefinition);
+                //If we're not active we don't need to check our requirements
+                if (!definesHashSet.Contains(scriptDefineDefinition.Define))
+                {
+                    continue;
+                }
+                
+                //Ensure that our requirements are present in case some other script or the developer manually removed them
+                foreach (string define in scriptDefineDefinition.RequiredDefines.Where(define => !definesHashSet.Contains(define)))
+                {
+                    LOGGER.Error($"{scriptDefineDefinition.Define} requires that {define} is set, but it's not present! Removing {scriptDefineDefinition.Define} and all dependencies if they were set.");
+                    RemoveDependentDefines(scriptDefineDefinition, definesHashSet);
+                    PlayerSettings.SetScriptingDefineSymbols(NAMED_BUILD_TARGET, definesHashSet.ToArray());
```

Comment:
Should this be outside the `foreach` loops?
Otherwise, you're potentially repeatedly setting the define symbols in each iteration of the loop

Reply:
Good catch. Updated